Deprecate SHA1 based algorithms for SAML signatures (#1739)
keycloak/keycloak-documentation: upgrading/topics/keycloak/changes-21_0_0.adoc

|`agroal_*`

|===

= Deprecated RSA_SHA1 and DSA_SHA1 algorithms for SAML

> Algorithms `RSA_SHA1` and `DSA_SHA1`, which can be configured as `Signature algorithms` on SAML adapters, clients and identity providers are deprecated. It is recommended to use safer
> alternatives based on `SHA256` or `SHA512`. Other reason is, that verifying signatures on signed SAML documents or assertions with these
> algorithms does not work on Java 17 or later. So in case that you use this algorithm and the other party consuming your SAML documents is running on Java 17 or later, verifying signatures will not work.

Reviewer: ```suggestion
Algorithms `RSA_SHA1` and `DSA_SHA1`, which can be configured as `Signature algorithms` on SAML adapters, clients and identity providers are deprecated. We recommend to use safer
alternatives based on `SHA256` or `SHA512`. Also, verifying signatures on signed SAML documents or assertions with these
algorithms do not work on Java 17 or higher. If you use this algorithm and the other party consuming your SAML documents is running on Java 17 or higher, verifying signatures will not work.
```

> The possible workaround is to remove algorithms like `http://www.w3.org/2000/09/xmldsig#rsa-sha1` or `http://www.w3.org/2000/09/xmldsig#dsa-sha1` from the list

Reviewer: ```suggestion
The possible workaround is to remove algorithms such as `http://www.w3.org/2000/09/xmldsig#rsa-sha1` or `http://www.w3.org/2000/09/xmldsig#dsa-sha1` from the list
```
